common/src/main/java/com/microsoft/identity/common/internal/broker/InactiveBrokerClient.java

```diff
@@ -0,0 +1,166 @@
+            if (resultBundle == null){
+                Logger.verbose(TAG + methodName, "resultBundle is null");
+                return null;
+            }
+
+            return resultBundle.getString(AuthenticationConstants.Broker.BROKER_KEYSTORE_SYMMETRIC_KEY);
+
+        } catch (final BaseException | InterruptedException | ExecutionException | RemoteException e) {
+            Logger.error(
+                    TAG + methodName,
+                    "Exception is thrown when trying to get key from inactive broker:"
+                            + e.getMessage(),
+                    e);
+        } finally {
+            client.disconnect();
+        }
+
+        return null;
+    }
+
+    /**
+     * Constructor for the BrokerAccountServiceClient
+     *
+     * @param context
+     * @param callingPackageName name of the calling package.
+     */
+    private InactiveBrokerClient(@NonNull final Context context,
+                                 @NonNull final String callingPackageName) {
+        mContext = context;
+        mCallingPackageName = callingPackageName;
+        mBrokerAccountServiceIntent = getIntentForBrokerAccountService(mContext);
+    }
+
+    /**
+     * Binds to the service and returns a future that provides the proxy for the calling the BrokerAccountService
+     *
+     * @return MicrosoftAuthServiceFuture
+     */
+    private BrokerAccountServiceFuture connect() throws ClientException {
+
+        if (mBrokerAccountServiceIntent == null) {
+            throw new ClientException("mBrokerAccountServiceIntent is null. BrokerAccountService.");
+        }
+
+        BrokerAccountServiceFuture future = new BrokerAccountServiceFuture();
+        mBrokerAccountServiceConnection = new BrokerAccountServiceConnection(future);
+
+        mBound = mContext.bindService(mBrokerAccountServiceIntent, mBrokerAccountServiceConnection, Context.BIND_AUTO_CREATE);
+        Logger.verbose(TAG + "connect", "The status for MicrosoftAuthService bindService call is: " + Boolean.valueOf(mBound));
+
+        if (!mBound) {
+            throw new ClientException("Service is unavailable or does not support binding. BrokerAccountService.");
+        }
+
+        return future;
+    }
+
+    /**
+     * Disconnects (unbinds) from the bound BrokerAccountService
+     */
+    private void disconnect() {
+        if (mBound) {
+            mContext.unbindService(mBrokerAccountServiceConnection);
+            mBound = false;
+        }
+    }
+
+    /**
+     * Gets the intent that points to the bound service on the device... if available
+     * You shouldn't get this far if it's not available
+     *
+     * @param context
+     * @return Intent
+     */
+    private Intent getIntentForBrokerAccountService(final Context context) {
+        final String inactiveBrokerPackageName = getInactiveBrokerPackageName(context);
+
+        if (inactiveBrokerPackageName == null || inactiveBrokerPackageName.length() == 0) {
```

Comment:
can be simplified to `TextUtils.isEmpty(inactiveBrokerPackageName)`

Reply:
@rpdome unresolving this as this is still not fixed after the `Resolve comments` commit